### ISSUE TITLE
Support multilabel ground truth

### ISSUE THREAD

**Issue by christianabbet** [swisstopo/swissgeol-boreholes-dataextraction#485]
Currently, each sample is assigned a single ground truth label. 

# Proposed change

Extend the data pipeline, training, and evaluation to support multilabel ground truth, where a single sample can be associated with more than one class simultaneously.

# Scope

* Update reduce_label (and callers) to return a list of labels instead of a single string
* Adapt the dataset/collation logic to produce multilabel targets (e.g. multi-hot vectors)
* Switch the loss function to one compatible with multilabel classification (e.g. BCEWithLogitsLoss)

# Example data 
#411  Organic components (unconsolidated rock) can habe multiple values, e.g. 

```
"organic_components": [
                            "humus",
                            "roots"
                        ],
```
